test/integration/graphql_test.lua

```diff
@@ -1159,3 +1161,142 @@ function g.test_both_data_and_error_result()
         {message = 'Simple error B'},
     })
 end
+
+function g.test_introspection()
+    local function callback(_, _)
+        return nil
+    end
+
+    local query_schema = {
+        ['test'] = {
+            kind = types.string.nonNull,
+            arguments = {
+                arg = types.string.nonNull,
+                arg2 = types.string,
+                arg3 = types.int,
+                arg4 = types.long,
+            },
+            resolve = callback,
+        }
+    }
+
+    local mutation_schema = {
+        ['test_mutation'] = {
+            kind = types.string.nonNull,
+            arguments = {
+                arg = types.string.nonNull,
+                arg2 = types.string,
+                arg3 = types.int,
+                arg4 = types.long,
+            },
+            resolve = callback,
+        }
+    }
+
+    local directives = {
+        types.directive({
+            name = 'custom',
+            arguments = {},
+            onQuery = true,
+            onMutation = true,
+            onField = true,
+            onFragmentDefinition = true,
+            onFragmentSpread = true,
+            onInlineFragment = true,
+            onVariableDefinition = true,
+            onSchema = true,
+            onScalar = true,
+            onObject = true,
+            onFieldDefinition = true,
+            onArgumentDefinition = true,
+            onInterface = true,
+            onUnion = true,
+            onEnum = true,
+            onEnumValue = true,
+            onInputObject = true,
+            onInputFieldDefinition = true,
+            isRepeatable = true,
+        })
+    }
+
+    local data, errors = check_request(introspection.query, query_schema, mutation_schema, directives)
+    t.assert_type(data, 'table')
+    t.assert_equals(errors, nil)
+end
+
+function g.test_custom_directives()
+    local function callback(_, _, info)
+        return require('json').encode(info.directives)
+    end
+
+    local query_schema = {
+        ['prefix'] = {
+            kind = types.object({
+                name = 'prefix',
+                fields = {
+                    ['test'] = {
+                        kind = types.string,
+                        arguments = {
+                            arg = types.string.nonNull,
+                            arg2 = types.string,
+                            arg3 = types.int,
+                            arg4 = types.long,
+                        },
+                        resolve = callback,
+                    }
+                },
+            }),
+            arguments = {},
+            resolve = function()
+                return {}
+            end,
+        }
+    }
+
+    local directives = {
+        types.directive({
+            name = 'custom',
+            arguments = {
+                arg = types.string.nonNull,
```

Comment:
I guess we need to test more complex scenario when directive type has inputObject type (see https://github.com/tarantool/tdg2/issues/1322)

Reply:
Added new test case